File: packages/java/src/main/java/dev/kreuzberg/literllm/BudgetExceededException.java
package dev.kreuzberg.literllm;

/**
 * Thrown when a request would exceed the configured cost budget.
 *
 * @see BudgetConfig
 */
public final class BudgetExceededException extends LlmException {

	/** Error code for budget exceeded errors. */
	public static final int CODE_BUDGET_EXCEEDED = 1800;

	/**
	 * Creates a budget-exceeded exception.
	 *
	 * @param message
	 *            human-readable description of the budget violation
	 */
	public BudgetExceededException(String message) {
		super(CODE_BUDGET_EXCEEDED, "liter-llm: budget exceeded: " + message);
	}
}

File: packages/php/src/BudgetExceededException.php
<?php

declare(strict_types=1);

namespace LiterLlm;

/**
 * Thrown when a request would exceed the configured cost budget.
 *
 * @see BudgetConfig
 */
final class BudgetExceededException extends \RuntimeException
{
    public function __construct(string $message, ?\Throwable $previous = null)
    {
        parent::__construct('liter-llm: budget exceeded: ' . $message, 1800, $previous);
    }
}
